host-bin/startxiwi

```diff
@@ -0,0 +1,93 @@
+  -k KEYFILE  Override the auto-detected encryption key location.
+  -n NAME     Name of the chroot to enter. Default: first one found in CHROOTS
+  -t TARGET   Only enter the chroot if it contains the specified TARGET.
+  -u USERNAME Username (or UID) to log into. Default: 1000 (the primary user)
+  -X XMETHOD  (Override the auto-detected XMETHOD for this session.) - deprecated, 'xiwi' is enforced.
+  -x          Does not log in, but directly executes the command instead.
+              Note that the environment will be empty (sans TERM).
+              Specify -x a second time to run the /prepare.sh script.
+"
+USAGE_XIWI="xiwi [options] CHROOT_APP [options]
+
+  -f          Prevent xiwi from quitting automatically.[1]
+  -F          Will launch the app full-screen.[2]
+  -T          Will launch the app in a tab.
+
+By default, the CHROOT_APP is launched in a window.
+
+  [1] xiwi will normally close when the application returns. Some gui applications
+      fork before or during normal operation, which can confuse xiwi and cause it to
+      quit prematurely. If your application does not have a parameter that prevents
+      it from forking, and crouton is unable to automatically detect the fork, you can
+      use -f to prevent xiwi from quitting automatically.
+      xiwi will quit if you close the Chromium OS window when nothing is displayed.
+  
+  [2] A default window manager will full-screen all windows, unless CHROOT_APP begins
+      with 'start' or is 'xinit'. You can use -F to force the CHROOT_APP full-screen.
+
+  You can cycle through multiple windows inside the
+  application via Ctrl-Alt-Tab/Ctrl-Alt-Shift-Tab, or close them via
+  Ctrl-Alt-Shift-Escape.  If CHROOT_APP begins with 'start' but you still want to
+  use the default window manager, specify the full path of the application.
+"
+
+# Exits the script with exit code $1, spitting out message $@ to stderr
+error() {
+    local ecode="$1"
+    shift
+    echo "$*" 1>&2
+    exit "$ecode"
+}
+
+options=':bc:hlk:n:t:u:xfFHT'
+while getopts $options OPT; do
+  case ${OPT} in
+    b) OPTS_ENTER="$OPTS_ENTER -b";;
+    c) OPTS_ENTER="$OPTS_ENTER -c ${OPTARG}";;
+    h) error 0 "$USAGE_ENTER";;
+    l) OPTS_ENTER="$OPTS_ENTER -l";;
+    k) OPTS_ENTER="$OPTS_ENTER -k ${OPTARG}";;
+    n) OPTS_ENTER="$OPTS_ENTER -n ${OPTARG}";;
+    t) OPTS_ENTER="$OPTS_ENTER -t ${OPTARG}";;
+    u) OPTS_ENTER="$OPTS_ENTER -u ${OPTARG}";;
+    x) OPTS_ENTER="$OPTS_ENTER -x";;
+    f) OPTS_XIWI="$OPTS_XIWI -f";;
+    F) OPTS_XIWI="$OPTS_XIWI -F";;
+    H) error 0 "$USAGE_XIWI";;
+    T) OPTS_XIWI="$OPTS_XIWI -T";;
+   \?) echo "Unknown option: -$OPTARG" 1>&2
+       error 2 "$USAGE";;
+    :) echo "Missing option argument for -$OPTARG"
+       error 2 "$USAGE";;
+    *) echo "Unimplemented option: -$OPTARG" 1>&2
+       error 2 "$USAGE";;
+  esac
+done
+shift "$((OPTIND-1))"
+
+sh -e "`dirname "\`readlink -f "$0"\`"`/enter-chroot" $OPTS_ENTER -t xiwi xiwi $OPTS_XIWI "$@"
```

Comment:
Also, need more exec's.  The whole line should be an exec, and prior to "xiwi" there should be an exec.